glean-core/src/database/mod.rs

```diff
@@ -274,6 +274,11 @@ impl Database {
 
     /// Records a metric in the underlying storage system.
     pub fn record(&self, glean: &Glean, data: &CommonMetricData, value: &Metric) {
+        // If upload is disabled we don't want to record.
+        if !glean.is_upload_enabled() {
```

Comment:
Random thought -- when we disable telemetry, we set the client-id to `c0ffee`.  Does that still work?

Reply:
Interesting, let me check.

Reply:
Should still work, TIL about this behaviour, but we set the client-id to `c0ffee` on the `clear_metrics` function, which [is called before we set `upload_enabled` to `false`](https://github.com/mozilla/glean/blob/main/glean-core/src/lib.rs#L418-L419)*.

\* At first sight that is what we do, but actually from looking closer at the code we set `upload_enabled` to `false` last thing on the `clear_metrics` function  and we assume `upload_enabled` is already `false` upon entering this function (see: https://github.com/mozilla/glean/blob/main/glean-core/src/lib.rs#L457-L478). Still, the changes in this PR don't affect this behaviour.